Libraries/LibWeb/Page/Page.cpp

```diff
@@ -637,14 +638,14 @@ Page::FindInPageResult Page::perform_find_in_page_query(FindInPageQuery const& q
         }
     }
 
-    if (direction.has_value()) {
-        if (direction.value() == SearchDirection::Forward) {
+    if (direction.has_value() && should_update_match_index) {
+        if (direction == SearchDirection::Forward) {
             if (m_find_in_page_match_index >= all_matches.size() - 1) {
                 if (query.wrap_around == WrapAround::No)
                     return {};
                 m_find_in_page_match_index = 0;
             } else {
-                m_find_in_page_match_index++;
+                m_find_in_page_match_index = m_find_in_page_match_index + 1;
```

Comment:
:thinking: why the change from `++m_find_in_page_match_index`?

Reply:
Yeah, this seems unnecessary. Reverted.